src/lib/node.ts

```diff
@@ -231,45 +231,50 @@ type FromNodeReadableDataType<T> = T extends void ? string | Buffer : T;
  * @param nodeReadable Soource NodeJS stream
  * @returns Readable stream of type `string | Buffer`
  */
-export function fromNodeReadable<T = void>(
-	nodeReadable: NodeJS.ReadableStream
-): ReadableStream<FromNodeReadableDataType<T>> {
-	const stream = new Stream<FromNodeReadableDataType<T>>();
+export function fromNodeReadable<T = string | Buffer>(
+	nodeReadable: NodeStream.Readable
+): ReadableStream<T> {
+	let stream = new Stream<T>();
 
 	const endStream = async (err?: Error) => {
 		if (!stream.isEndingOrEnded()) {
-			/**
-			 * `stream.end` could reject, so we'll emit
-			 * any caught errors as an event via the NodeJS stream.
-			 * This is acceptable as we are only attempting to end
-			 * the ts-stream a maximum one time.
-			 */
 			try {
-				return stream.end(err);
+				await stream.end(err);
 			} catch (e) {
-				nodeReadable.emit("error", e);
+				nodeReadable.destroy(ensureError(e));
 			}
 		}
 	};
 
-	stream.aborted().catch(endStream);
-	nodeReadable.once("close", endStream);
-	nodeReadable.once("error", endStream);
+	const handleError = (err: unknown) => {
+		endStream(ensureError(err));
+		nodeReadable.destroy(ensureError(err));
+	};
+
+	stream.aborted().catch(handleError);
+	nodeReadable.once("error", handleError);
+	nodeReadable.once("close", () => endStream());
```

Comment:
Can this be just `endStream` instead of the extra closure? (Not sure.)

Reply:
It could be, but I wanted to make sure that we are not calling `endStream` with any arguments. Although, the documentation doesn't specify any function arguments for the callback https://nodejs.org/api/stream.html#stream_event_close_1

Reply:
Hmyeah, makes sense, especially because endStream indeed could receive an argument. Ok, let's keep it!